docs/js-core-2/cowsay-project/readme.md

```diff
@@ -0,0 +1,120 @@
+
+```
+npm install cowsay
+npx cowsay Moooo
+```
+
+We did it! But let's explore some more. How would you install cowsay as a dependency? Make a new folder called /cowsaying and initialise a new package.
+```
+mkdir /cowsaying
+cd /cowsaying
+npm init
+```
+Press return through all the options and then do.
+```
+npm install cowsay
+```
+### npm run
+
+Let's find out how to execute cowsay from our package.json. Open /cowsaying in your text editor and look at the [package.json](https://nodejs.dev/learn/the-package-json-guide).
+
+Add cowsay to the scripts array. 
+```
+"scripts": {
+    "cowsay": "cowsay Mooooo"
+  },
+```
+Now cowsay is listed as a script, and it has an argument.  In your terminal, run cowsay.
+```
+npm run cowsay
+```
+What are the downsides of this approach? I would say one is that we want to be able to pass in arguments, not just always say moo. How could we make the cow say Baaaa?
+
+Remove the hardcoded "Mooooo".  Can you pass a different string in the command line?
+
+### node 
+
+Now let's execute cowsay in a different way. Open /index.js in your text editor. Import the cowsay package with require.
+```
+const cowsay = require("cowsay");
+```
+And log a cow to the console that says "I require a package". Add some more arguments if you can.
+```
+console.log(cowsay.say({
+        text : "I require a package",
+        e : "oO",
+        T : "U "
+    }));
+```
+Now run this with node:
+```
+node index.js
+```
+## Going deeper, your stretch goal
+
+How is this cow being made and what else can we make? Open your node modules and look inside cowsay. There's a README file that tells you all about the package and a library of other animals you can make. Read it. You don't have to understand all the code to get an idea of what is going on. 
+
+How would you extend this library to add a new animal? What about [a duck](https://www.asciiart.eu/animals/birds-water)? Look at how the existing ```.cows``` are imported and try adapting this for your own .cow. There are no solutions provided for this challenge. Dig around for yourself.
+
+## Project
+
+Let's figure out how to make a cow say things in Node by ourselves. For this project we don't really need a package, a library or lots of options. Let's just get the cow printing out and saying whatever we write in the command line. What would be helpful? I think we need to:
+
+* [Accept an argument](https://nodejs.dev/learn/nodejs-accept-arguments-from-the-command-line) from the command line.
+* Output to the command line. You've already done this with console.log.
+* Make an ASCII cow.
+* Write a function that puts the string into the cow's speech bubble.
+
+Write your solution in a file called solution.js and test it by running your program in the command line. How will you handle a null argument?
```

Comment:
I think there are two layers being discussed here - the sentence is about the command line (where you can't pass null), and the "null argument" question seems to be more about the function... It could be a little clearer to say something like "no argument being passed" so that both sentences are talking about the same layer of abstraction?